src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -1168,6 +1178,67 @@ void Qgs3DMapCanvasWidget::setSceneExtent( const QgsRectangle &extent )
     mMainCanvas->unsetMapTool( mMapToolExtent.get() );
 }
 
+void Qgs3DMapCanvasWidget::setClippingPlanesOn2DCanvas()
+{
+  if ( !qobject_cast<QgsMapToolClippingPlanes *>( mMainCanvas->mapTool() ) )
+    mMapToolPrevious = mMainCanvas->mapTool();
+
+  mMainCanvas->setMapTool( mMapToolClippingPlanes.get() );
+  QgisApp::instance()->activateWindow();
+  QgisApp::instance()->raise();
+  mMessageBar->pushInfo( QString(), tr( "Select a rectangle by 3 points on the main 2D map view to define this 3D scene's cross section" ) );
+}
+
+void Qgs3DMapCanvasWidget::setViewOnClippingPlanesChanged( const QList<QVector4D> &planes )
+{
+  this->activateWindow();
+  this->raise();
+  mMessageBar->clearWidgets();
+
+  // calculate the middle of the front side defined by clipping planes with Cramer's rule
+  double det = planes.at( 0 ).x() * planes.at( 3 ).y() - planes.at( 3 ).x() * planes.at( 0 ).y();
+  const QgsVector3D point1(
+    ( planes.at( 0 ).w() * planes.at( 3 ).y() - planes.at( 3 ).w() * planes.at( 0 ).y() ) / -det,
+    ( planes.at( 3 ).w() * planes.at( 0 ).x() - planes.at( 0 ).w() * planes.at( 3 ).x() ) / -det,
+    0
+  );
+  det = planes.at( 2 ).x() * planes.at( 3 ).y() - planes.at( 3 ).x() * planes.at( 2 ).y();
+  const QgsVector3D point2(
+    ( planes.at( 2 ).w() * planes.at( 3 ).y() - planes.at( 3 ).w() * planes.at( 2 ).y() ) / -det,
+    ( planes.at( 3 ).w() * planes.at( 2 ).x() - planes.at( 2 ).w() * planes.at( 3 ).x() ) / -det,
+    0
+  );
+  QgsVector3D middle( point1 + ( point2 - point1 ) / 2 );
+  // we temporarily disable terrain so the elevation calculation won't take it in account if the terrain is flat
+  const bool lastTerrainState = mCanvas->scene()->mapSettings()->terrainRenderingEnabled();
+  if ( dynamic_cast<const QgsFlatTerrainSettings *>( mCanvas->scene()->mapSettings()->terrainSettings() ) )
+  {
+    mCanvas->scene()->mapSettings()->setTerrainRenderingEnabled( false );
+  }
+  const double side = std::max( middle.distance( point1 ), ( mCanvas->scene()->elevationRange().upper() - mCanvas->scene()->elevationRange().lower() ) / 2 );
```

Comment:
I think that It should be possible to add a `ignoreTerrain` parameter to `mCanvas->scene()->elevationRange()` to ignore the terrain and avoid disabling the terrain here.